Docker Desktop 4.19.0 release notes (#17167)
docker/docs: desktop/release-notes.md

For frequently asked questions about Docker Desktop releases, see [FAQs](faqs/general.md/#releases)

## 4.19.0

{% include release-date.html date="2023-04-27" %}

{% include desktop-install.md all=true version="4.19.0" build_path="/" %}

### New

TODO: Work in progress

### Upgrades

- [Docker Engine v23.0.3](https://docs.docker.com/engine/release-notes/23.0/#2303)
- [Compose 2.17.3](https://github.com/docker/compose/releases/tag/v2.17.3)
- [Containerd v1.6.20](https://github.com/containerd/containerd/releases/tag/v1.6.20)
- [Kubernetes v1.26.3](https://github.com/kubernetes/kubernetes/releases/tag/v1.26.3)
- [runc v1.1.5](https://github.com/opencontainers/runc/releases/tag/v1.1.5)
- [criTools v1.26.1](https://github.com/kubernetes-sigs/cri-tools/releases/tag/v1.26.1)
- [cri-dockerd v0.3.1](https://github.com/Mirantis/cri-dockerd/releases/tag/v0.3.1)
- [coredns v1.9.3](https://github.com/coredns/coredns/releases/tag/v1.9.3)
- [cni v1.2.0](https://github.com/containernetworking/plugins/releases/tag/v1.2.0)
- [Go v1.20.3](https://github.com/golang/go/releases/tag/go1.20.3)

### Bug fixes and enhancements

#### For all platforms

> - The `docker scan` command has been removed. To continue learning about the vulnerabilities of your images, and many other features, use the new `docker scout` command. Run `docker scout --help`, or learn more at https://docs.docker.com/engine/reference/commandline/scout/.

Reviewer: We normally have a separate section for removals / deprecations. We give them more prominence because people actually need to go and update their workflows/scripts. (See 4.3.0 & 4.10.0 for examples).

### New

TODO: Work in progress

### Removed

- Removed `docker scan` command. To continue learning about the vulnerabilities of your images, and many other features, use the new `docker scout` command. Run `docker scout --help`, or learn more at https://docs.docker.com/engine/reference/commandline/scout/.

### Upgrades

- [Docker Engine v23.0.3](https://docs.docker.com/engine/release-notes/23.0/#2303)
- [Compose 2.17.3](https://github.com/docker/compose/releases/tag/v2.17.3)
- [Containerd v1.6.20](https://github.com/containerd/containerd/releases/tag/v1.6.20)
- [Kubernetes v1.26.3](https://github.com/kubernetes/kubernetes/releases/tag/v1.26.3)
- [runc v1.1.5](https://github.com/opencontainers/runc/releases/tag/v1.1.5)
- [criTools v1.26.1](https://github.com/kubernetes-sigs/cri-tools/releases/tag/v1.26.1)
- [cri-dockerd v0.3.1](https://github.com/Mirantis/cri-dockerd/releases/tag/v0.3.1)
- [coredns v1.9.3](https://github.com/coredns/coredns/releases/tag/v1.9.3)
- [cni v1.2.0](https://github.com/containernetworking/plugins/releases/tag/v1.2.0)
- [Go v1.20.3](https://github.com/golang/go/releases/tag/go1.20.3)

### Bug fixes and enhancements

> #### For all platforms

Reviewer 2: Per https://docs.google.com/document/d/1Ndjvbwr1GXbve2GOn0_laVh9tMAM-_zn02k0litRZGI/edit#
We should also include here "Docker init (Beta) adds support for Node.js and Python"

Author: This will most likely go into the `### New` section once the highlights are finalized.

- Improved `docker scout compare` command to compare two images, now also aliased under `docker scout diff`.

> - Added in-app walkthroughs to Learning Center.

Author: @aevesdocker Currently the highlights contains `Learning center now supports in-product walkthroughs` which will go into the `New` section once finalized, maybe we should get rid of this entry here then? wdyt?

> - Added in-app walkthroughs to Learning Center.

Reviewer 3: yes agreed!

- Added more details to dashboard errors when a `docker-compose` action fails ([docker/for-win#13378](https://github.com/docker/for-win/issues/13378)).

> - Added support for setting HTTP proxy configuration during installation. This can be done via the `--proxy-http-mode`, `--overrider-proxy-http`, `--override-proxy-https` and `--override-proxy-exclude` installer flags in the case of installation from the CLI on [Mac](../install/mac-install.md#install-from-the-command-line) and [Windows](../install/windows-install.md#install-from-the-command-line), or alternatively by setting the values in the `install-settings.json` file.

Reviewer 3: ```suggestion
- Added support for setting HTTP proxy configuration during installation. This can be done via the `--proxy-http-mode`, `--overrider-proxy-http`, `--override-proxy-https` and `--override-proxy-exclude` installer flags in the case of installation from the CLI on [Mac](/install/mac-install.md#install-from-the-command-line) and [Windows](/install/windows-install.md#install-from-the-command-line), or alternatively by setting the values in the `install-settings.json` file.
```

- Stop overriding .docker/config.json `credsStore` keys on application start. Note that if the developer uses a custom credential helper then the CLI `docker login` and `docker logout` will not affect whether the UI is signed in to Docker or not. In general it is better to sign into Docker via the UI since the UI supports multi-factor authentication.
- Added a warning about the [forthcoming removal of Compose V1 from Docker Desktop](https://docs.docker.com/go/compose-v1-eol/). Can be suppressed with `COMPOSE_V1_EOL_SILENT=1`.
- In Compose config, boolean fields in YAML should be either `true` or `false`. Deprecated YAML 1.1 values such as “on” or “no” will now produce a warning.
- Improved UI for image table, allowing rows to use more available space.
- Fixed various bugs in port-forwarding.
- Fixed a HTTP proxy bug where an HTTP request without a Server Name Indication record would be rejected with an error.

#### For Windows

> - Reverted to fully patching etc/hosts on windows (includes `host.docker.internal` and `gateway.docker.internal` again). For WSL, this behavior is controlled by a new General setting.

Reviewer 3: ```suggestion
- Reverted to fully patching etc/hosts on windows (includes `host.docker.internal` and `gateway.docker.internal` again). For WSL, this behavior is controlled a new setting in the **General** tab. 
```

  Fixes [docker/for-win#13388](https://github.com/docker/for-win/issues/13388) and [docker/for-win#13398](https://github.com/docker/for-win/issues/13398).
- Fixed a spurious `courgette.log` file appearing on the Desktop when updating Docker Desktop. Fixes [docker/for-win#12468](https://github.com/docker/for-win/issues/12468).

> - Fixed "zoom in" shortcut (ctrl+=). Fixes [docker/for-win#13392](https://github.com/docker/for-win/issues/13392).

Reviewer 3: ```suggestion
- Fixed the "zoom in" shortcut (ctrl+=). Fixes [docker/for-win#13392](https://github.com/docker/for-win/issues/13392).
```

- Fixed a bug where the tray menu would not correctly update after second container type switch. Fixes [docker/for-win#13379](https://github.com/docker/for-win/issues/13379).

#### For Mac

- Increase performance of VM networking when using Virtualization framework on macOS Ventura and above.
- Fixed a bug where an error window is displayed on uninstall.
- Fixed a bug where the setting `deprecatedCgroupv1` was ignored.
- Fixed cases where `docker pull` would return `EOF`.

#### For Linux

- Fixed a bug where the VM networking crashes after 24h. Fixes [docker/desktop-linux#131](https://github.com/docker/desktop-linux/issues/131).

### Security

#### For all platforms

> - Fixed a security issue in Artifactory Integration where it would fallback to sending registry credentials over plain HTTP if HTTPS check failed. Fixes [docker/for-win#13344](https://github.com/docker/for-win/issues/13344).

Reviewer 3: ```suggestion
- Fixed a security issue in the Artifactory Integration where it would fallback to sending registry credentials over plain HTTP if HTTPS check failed. Fixes [docker/for-win#13344](https://github.com/docker/for-win/issues/13344).
```

For frequently asked questions about Docker Desktop releases, see [FAQs](faqs/general.md/#releases)

## 4.19.0

{% include release-date.html date="2023-04-27" %}

> {% include desktop-install.md all=true version="4.19.0" build_path="/" %}

Reviewer 3: Guessing the build path will be updated last minute? @maxcleme

- Fixes [docker/for-win#13388](https://github.com/docker/for-win/issues/13388) and [docker/for-win#13398](https://github.com/docker/for-win/issues/13398).
- Fixed a spurious `courgette.log` file appearing on the Desktop when updating Docker Desktop. Fixes [docker/for-win#12468](https://github.com/docker/for-win/issues/12468).
- Fixed the "zoom in" shortcut (ctrl+=). Fixes [docker/for-win#13392](https://github.com/docker/for-win/issues/13392).
- Fixed a bug where the tray menu would not correctly update after second container type switch. Fixes [docker/for-win#13379](https://github.com/docker/for-win/issues/13379).

#### For Mac

- Increased the performance of VM networking when using Virtualization framework on macOS Ventura and above.
- Fixed a bug where an error window is displayed on uninstall.
- Fixed a bug where the setting `deprecatedCgroupv1` was ignored.
- Fixed cases where `docker pull` would return `EOF`.

#### For Linux

- Fixed a bug where the VM networking crashes after 24h. Fixes [docker/desktop-linux#131](https://github.com/docker/desktop-linux/issues/131).

### Security

#### For all platforms

> - Fixed a security issue in the Artifactory Integration where it would fallback to sending registry credentials over plain HTTP if HTTPS check failed. Fixes [docker/for-win#13344](https://github.com/docker/for-win/issues/13344).

Reviewer 4: @maxcleme This is already in the 4.18 release notes, is there any reason for including it here?

Author: @gabriellavengeo Nice catch, I believe this one was cherry picked in 4.18.0, that is explaining why it appears on both. I'll a do a quick PR to get rid of it, thanks! 🙇

For frequently asked questions about Docker Desktop releases, see [FAQs](faqs/general.md/#releases)

## 4.19.0

{% include release-date.html date="2023-04-27" %}

{% include desktop-install.md all=true version="4.19.0" build_path="/" %}

### New

- Docker Engine and CLI updated to [Moby 23.0](https://github.com/moby/moby/releases/tag/v23.0.0).

> - The `docker scan` command has been removed, please use `docker scout cves`.

Reviewer 3: I'd remove this. already covered in the 'Removed' section

> - Learning Center now supports in-product walkthroughs.

Reviewer 3: ```suggestion
- The **Learning Center** now supports in-product walkthroughs.
```

> - Docker init (Beta) adds support for Node.js and Python.

Reviewer 3: ```suggestion
- Docker init (Beta) now supports Node.js and Python.
```

- Faster networking between VM and host on macOS.
- You can now inspect and analyze remote images from Docker Desktop without pulling them.

> - Usability and performance improvements to the Artifactory images view.

Reviewer 3: ```suggestion
- Usability and performance improvements to the **Artifactory images** view.
```
